### PR TITLE
OOP solution

### PR DESCRIPTION
**Pull request by z1row**

### CODE REVIEW

File: src/main/java/core/basesyntax/Circle.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    private final int radius;
+
+    public Circle(int radius, String color) {
+        this.radius = radius;
+        this.color = color;
+    }
+
+    @Override
+    public double getArea() {
+        return Math.PI * radius;
```
Comment (olekskov) [inline src/main/java/core/basesyntax/Circle.java:13]: seems circle area has slightly another formula

File: src/main/java/core/basesyntax/Color.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public enum Color {
+    black, white, yellow, green, pink, red, brown, purple
```
Comment (olekskov) [inline src/main/java/core/basesyntax/Color.java:4]: enum elements naming follows constants naming convention

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        final int index = random.nextInt(5);
```
Comment (olekskov) [inline src/main/java/core/basesyntax/FigureSupplier.java:10]: this and all below magic values should be constants with proper naming

File: src/main/java/core/basesyntax/Main.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        Figure[] figures = new Figure[6];
```
Comment (olekskov) [inline src/main/java/core/basesyntax/Main.java:6]: also magic numbers shouldn't be inlined, use constant

File: src/main/java/core/basesyntax/RightTriangle.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
+    private final int side;
```
Comment (olekskov) [inline src/main/java/core/basesyntax/RightTriangle.java:4]: I don't fully understand how we may describe right triangle with one side.
Seems we need 2 sides that create right corner

File: src/main/java/core/basesyntax/RightTriangle.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+    @Override
+    public void draw() {
+        System.out.println("Figure: right triangle, area: " + String.format("%.2f", getArea())
+                + ", side: " + side + ", color: " + color);
+    }
+
+    @Override
+    public double getArea() {
+        return (Math.sqrt(3) / 4) * Math.pow(side, 2);
```
Comment (olekskov) [inline src/main/java/core/basesyntax/RightTriangle.java:19]: after we have 2 sides area will be half of rectangle

**Review by olekskov — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/FigureSupplier.java (hunk 3 above)
Comment (z1row) [reply src/main/java/core/basesyntax/FigureSupplier.java:10]: image
image
i made it constants and got this

**Pull request closed without merging**